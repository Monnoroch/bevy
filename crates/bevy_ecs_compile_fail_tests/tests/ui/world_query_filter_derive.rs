use bevy_ecs::prelude::*;
use bevy_ecs::query::WorldQueryFilter;

#[derive(Component)]
struct Foo;

#[derive(WorldQueryFilter)]
struct ComponentQuery {
    a: &'static Foo,
}

fn main() {}
