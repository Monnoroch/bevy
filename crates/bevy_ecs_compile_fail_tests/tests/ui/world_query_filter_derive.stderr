error[E0277]: the trait bound `&'static Foo: WorldQueryFilter` is not satisfied
 --> tests/ui/world_query_filter_derive.rs:9:8
  |
9 |     a: &'static Foo,
  |        ^^^^^^^^^^^^ the trait `WorldQueryFilter` is not implemented for `&'static Foo`
  |
note: required by a bound in `assert_world_query_filter`
 --> tests/ui/world_query_filter_derive.rs:7:10
  |
7 | #[derive(WorldQueryFilter)]
  |          ^^^^^^^^^^^^^^^^ required by this bound in `assert_world_query_filter`
  = note: this error originates in the derive macro `WorldQueryFilter` (in Nightly builds, run with -Z macro-backtrace for more info)
